### PR TITLE
add throttle_step to Evalspec and  support estimator evaluation trigger with throttle_step

### PR DESCRIPTION
**Pull request by wenmengzhou**
try to implement throttle_step settings discussed in #17650，using throttle_step to trigger evaluation can be more robust to different GPU load。

### CODE REVIEW

**Comment by wenmengzhou** [comment]
I signed it

**Comment by wenmengzhou** [comment]
@xiejw  could you review this PR?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 90 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @case540: It has been 120 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

File: tensorflow/python/estimator/training.py [hunk 1]
```diff
@@ -771,13 +820,16 @@ def _start_continuous_evaluation(self):
     """Repeatedly calls `Estimator` evaluate and export until training ends."""
 
     _assert_eval_spec(self._eval_spec)
+    if self._eval_spec.throttle_secs is None:
```
Comment (ispirmustafa) [inline tensorflow/python/estimator/training.py]: For the sake of consistency (i.e. being able to use same config for distributed training or local one) please also support the throttle_steps here. You can use Estimator.get_variable_value(...) to get latest step value.

**Review by ispirmustafa — COMMENTED** [review]
Thank you Wenmeng.
Adding a new parameter has maintenance cost. I'm letting Karmel to decide whether it is worth it or not. Please wait her approval before putting more effort into it.

**Comment by rthadur** [comment]
@wenmengzhou can you please resolve conflicts

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**